core/trino-main/src/main/java/io/trino/server/TrinoSystemRequirements.java

```diff
@@ -82,8 +82,8 @@ else if ("ppc64le".equals(osArch)) {
             }
         }
         else if ("Mac OS X".equals(osName)) {
-            if (!"x86_64".equals(osArch)) {
-                failRequirement("Trino requires x86_64 on Mac OS X (found %s)", osArch);
+            if (!"x86_64".equals(osArch) && !"aarch64".equals(osArch)) {
+                failRequirement("Trino requires x86_64 or aarch64 on Mac OS X (found %s)", osArch);
```

Comment:
Mac is not supposed to be production-supported platform anyway, so we could as well remove this check when `"Mac OS X".equals(osName)`. "Warranty is void" anyway

Reply:
What does “production supported” actually mean? We run on macs all day long, so that, to me, means it’s supported.

This check is more about catching or excluding setups we’ve never even tried, in order to reduce the chances of user error and simplify troubleshooting.

Reply:
https://trino.io/docs/current/installation/deployment.html